### PR TITLE
Always create new process group if getpgrp(2) returns 0

### PR DESCRIPTION
**Pull request by mqudsi**
Another attempt at fixing #5295, this time putting fish in its own pgrp
when it is launched with pgrp 0 even if we are not interactive.

This should fix firejail invocations where $SHELL is `fish` (and not
just invocations where the target is `fish`, like in 6dafcc49605889385dfc3011f92eb6d25bbadeda).

### CODE REVIEW

**Comment by faho** [comment]
No joy. This makes the parent fish get SIGTTOU.

File: src/reader.cpp [hunk 1]
```diff
@@ -932,6 +932,28 @@ void reader_init() {
...
+    if (getpgrp() == 0) {
+        auto shell_pgid = getpid(); // the new pgid will be our own pid
+        if (setpgid(shell_pgid, shell_pgid) < 0) {
+            debug(0, _(L"Failed to assign shell to its own process group"));
+            wperror(L"setpgid");
+            exit_without_destructors(1);
+        }
+
+        // Take control of the terminal
+        if (tcsetpgrp(STDIN_FILENO, shell_pgid) == -1) {
```
Comment (faho) [inline src/reader.cpp]: This SIGTTOUs.

**Comment by mqudsi** [comment]
That's why I said firejail is intercepting the call to `getpgrp`. I wasn't able to test it on my end since installing firejail wanted to pull in a truckload of x11 dependencies I don't want on my headless machines, but I'll try building firejail from source to see if I can do so without the X11 garbage and give it a try.

**Comment by mqudsi** [comment]
I can't get firejail to work for me:

```
mqudsi@Blitzkrieg /m/d/r/fish> firejail --noprofile fish
Warning: an existing sandbox was detected. fish will run without any additional sandboxing features
Welcome to fish, the friendly interactive shell
mqudsi@Blitzkrieg /m/d/r/fish>
```

**Comment by faho** [comment]
This doesn't work, so let's close it.

**Pull request closed without merging**